CentOS 7 Support

petracvv opened the pull request:
I was trying to use this cookbook to install the opsmatic agent on CentOS 7, however it would still install the upstart package and try to use the upstart service provider. In addition, it was hard-coded to use the CentOS 6 opsmatic repository. 

I have made the following changes:
* Use the yum $releasever variable in the .repo file to support either CentOS 6 or CentOS 7
* Check for CentOS 7 in the agent recipe and install the opsmatic-agent-systemd package if found
* Check for CentOS 7 in the agent recipe and use the Chef systemd service provider if found
* Automatically restart the agent if a config file was updated by chef

Let me know if this looks good or if there is another approach you prefer.

mihasya commented:
If you have the ability to run kitchen tests, could you add a centos 7 platform to `.kitchen.yml` and make sure `kitchen converge` still passes for all the platforms? Thanks!

mihasya commented:
Also looks like one of the liters is unhappy - https://travis-ci.org/opsmatic/opsmatic-cookbook/builds/55540659 picky bastards :smile: 

You can run those tests locally by doing `bundle exec rake test`.

petracvv commented:
Ok. Fixed the linting checks (I am still far to used to using symbols for attributes it seems). 

Your kitchen file had a centos-6.4 entry which vagrant failed to find a download for; I updated that to centos 6.6 (the newest release). In addition, I added a 14.04 target to the file (My changes were working on our 14.04 machines internally) and a centos-7.0 target. 

All passed the kitchen tests locally.

mihasya commented:
:+1: Awesome! Thank you very much! I'll merge and bump the cookbook version.

One of these days, we'll hook up the kitchen tests to run in the cloud.
